app/routes/user/client_credentials.tsx

```diff
@@ -158,23 +232,112 @@ export default function Index() {
       body: JSON.stringify({ name, scope }),
     })
       .then((result) => result.json())
-      .then((item) => setItems([...items, item]))
+      .then((item) => {
+        setItems([item, ...items])
+        setSelectedTab(tabs()[0].index)
+      })
   }
 
+  function onChange(value: any) {
+    if (value) {
+      setDisableButton(false)
+    } else {
+      setDisableButton(true)
+    }
+  }
   return (
-    <>
+    <section>
       <h1>Client Credentials</h1>
       <div className="usa-prose">
         <p>
           A Client Credential is a randomly generated Client ID and Client
           Secret that you can use in script to connect to the GCN Kafka broker.
         </p>
       </div>
-      <p>
-        <ModalToggleButton modalRef={modalRef} opener>
-          Create new client credential
-        </ModalToggleButton>
-      </p>
+      <h3>Create New Client Credential</h3>
+      <section>
+        <div>
+          <div className="usa-prose">
+            <p id="modal-new-description">
+              Choose a name for your new client credential.
+            </p>
+            <p className="text-base">
+              The name should help you remember what you use the client
+              credential for, or where you use it. Examples: “My Laptop”, “Lab
+              Desktop”, “GRB Pipeline”.
+            </p>
+          </div>
+          <Alert
+            type="info"
+            validation
+            heading="Requirements"
+            headingLevel="h4"
+          >
+            <ValidationChecklist id="validate-code">
+              <ValidationItem id="name" isValid={validations.name}>
+                The Name field is required
+              </ValidationItem>
+            </ValidationChecklist>
+          </Alert>
+          <Label htmlFor="name">Name</Label>
+          <TextInput
+            data-focus
+            name="name"
+            id="name"
+            type="text"
+            placeholder="Name"
+            defaultValue={defaultName}
+            onChange={validateInput}
+          />
+          <Label htmlFor="scope">Scope</Label>
+          <Dropdown
+            id="scope"
+            name="scope"
+            defaultValue={defaultScope}
+            onChange={(e) => setScope(e.target.value)}
+            onBlur={(e) => setScope(e.target.value)}
+          >
+            {groups.map((group) => (
+              <option key={group} value={group}>
+                {group}
+              </option>
+            ))}
+          </Dropdown>
+          <br />
+          {process.env.NODE_ENV === 'production' ? (
+            <ReCAPTCHA sitekey={siteKey} onChange={onChange}></ReCAPTCHA>
```

Comment:
Don't we need to do some server-side checking of the captcha result? Otherwise the user could bypass the captcha simply by submitting the form programmatically.

Reply:
Updated the handleCreate to use the grecaptcha.getResponse method and just return when the result is an empty string